pkg/manager/member/pd_member_manager.go

```diff
@@ -214,7 +214,7 @@ func (m *pdMemberManager) syncPDStatefulSetForTidbCluster(tc *v1alpha1.TidbClust
 	}
 
 	// Force update takes precedence over scaling because force upgrade won't take effect when cluster gets stuck at scaling
-	if !tc.Status.PD.Synced && NeedForceUpgrade(tc.Annotations) {
+	if !tc.Status.PD.Synced && !templateEqual(newPDSet, oldPDSet) && (NeedForceUpgrade(tc.Annotations) || *oldPDSet.Spec.Replicas < 2) {
```

Comment:
We'd better move the replica check into the pd_upgrader.go, putting it here, after the first update, the tc is requeued, and then in the next loop, it will go to the upgrader logic in L243.

Reply:
The PD StatefulSet will be upgraded by UpdateStatefulSet and won't go to the upgrader logic in L243 again. In other hand, if PD crashLoopBackoff as #1241 said, it won't force upgrade PD and return directly: https://github.com/pingcap/tidb-operator/blob/3515c02b0cf74da1afdea1d827595b29cd0a6625/pkg/manager/member/pd_member_manager.go#L233-L238